From garzik@havoc.gtf.org Sat Mar  4 18:52:56 2006
Date: Sat, 4 Mar 2006 21:52:42 -0500
From: Jeff Garzik <jeff@garzik.org>
To: linux-kernel@vger.kernel.org
Cc: Greg KH <gregkh@suse.de>, akpm@osdl.org, torvalds@osdl.org
Subject: PCI: fix pci_request_region[s] arg
Message-ID: <20060305025242.GA11373@havoc.gtf.org>
Content-Disposition: inline

From: Jeff Garzik <jeff@garzik.org>

    Add missing 'const' to pci_request_region[s] 'res_name' arg,
    since we pass it directly to __request_region(), whose 'name' arg
    is also const.

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/pci.c   |    4 ++--
 include/linux/pci.h |    4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

--- gregkh-2.6.orig/drivers/pci/pci.c
+++ gregkh-2.6/drivers/pci/pci.c
@@ -648,7 +648,7 @@ void pci_release_region(struct pci_dev *
  *	Returns 0 on success, or %EBUSY on error.  A warning
  *	message is also printed on failure.
  */
-int pci_request_region(struct pci_dev *pdev, int bar, char *res_name)
+int pci_request_region(struct pci_dev *pdev, int bar, const char *res_name)
 {
 	if (pci_resource_len(pdev, bar) == 0)
 		return 0;
@@ -706,7 +706,7 @@ void pci_release_regions(struct pci_dev 
  *	Returns 0 on success, or %EBUSY on error.  A warning
  *	message is also printed on failure.
  */
-int pci_request_regions(struct pci_dev *pdev, char *res_name)
+int pci_request_regions(struct pci_dev *pdev, const char *res_name)
 {
 	int i;
 	
--- gregkh-2.6.orig/include/linux/pci.h
+++ gregkh-2.6/include/linux/pci.h
@@ -521,9 +521,9 @@ void pdev_sort_resources(struct pci_dev 
 void pci_fixup_irqs(u8 (*)(struct pci_dev *, u8 *),
 		    int (*)(struct pci_dev *, u8, u8));
 #define HAVE_PCI_REQ_REGIONS	2
-int pci_request_regions(struct pci_dev *, char *);
+int pci_request_regions(struct pci_dev *, const char *);
 void pci_release_regions(struct pci_dev *);
-int pci_request_region(struct pci_dev *, int, char *);
+int pci_request_region(struct pci_dev *, int, const char *);
 void pci_release_region(struct pci_dev *, int);
 
 /* drivers/pci/bus.c */
